src/cryptographic_primitives/commitments/hash_commitment.rs

```diff
@@ -8,34 +8,32 @@
 //TODO: (open issue) use this struct to represent the commitment HashCommitment{comm: BigInt, r: BigInt, m: BigInt}
 /// calculate commitment c = H(m,r) using SHA3 CRHF.
 /// r is 256bit blinding factor, m is the commited value
-pub struct HashCommitment;
+pub struct HashCommitment<H: Digest + Clone>(PhantomData<H>);
```

Comment:
```suggestion
pub struct HashCommitment<H: Digest + Clone>(PhantomData<fn(H)>);
```

Reply:
I didn't do that because none of the methods here accept `&self` so it doesn't matter if it's Send/Sync because you never Move it.
But we can decide to just always use `fn(H)` unless there's a good reason for something else

Reply:
Ah yes, you're right, I didn't notice that! It actually doesn't matter here